crates/core/src/context.rs

```diff
@@ -16,10 +18,15 @@ use marzano_language::target_language::TargetLanguage;
 use marzano_util::analysis_logs::AnalysisLogs;
 
 /// Contains various kinds of context about the query being executed.
-pub trait QueryContext: Clone + std::fmt::Debug + Sized {
+pub trait QueryContext: Clone + std::fmt::Debug + Sized + 'static {
```

Comment:
Why do we need to add the static lifetime here? I'm a little nervous it might make QueryContext overly inflexible